docs/3.2-Network-Config.md

```diff
@@ -1,48 +1,39 @@
-Shadow uses a graphml format to represent a network topology. The python module [networkx](http://networkx.github.io/) can be used to manipulate such a format. Shadow is distributed with some graphml topology files that were generated using data from a variety of sources. Because the topology is represented in a standard graphml format, it is easy to swap out specific measurements of latency, jitter, packetloss, etc, if desired.
+### Network Graph Overview
 
-### Vertices
+Processes running in Shadow do not have access to the internet; instead, processes running on Shadow virtual hosts utilize an internal routing module to communicate with other processes running on other virtual hosts in the simulation. The routing module is used to position virtual hosts within a network topology, to compute communication paths between virtual hosts, and to enforce network path characteristics like latency and packet loss.
 
-In addition to the default _id_ attribute, all vertices must have the _bandwidthup_ and _bandwidthdown_ attributes (given in KiB/s). Optional vertex attributes include _ip_, _citycode_, _countrycode_, and _packetloss_.
+This page describes the routing module and how it can be configured.
 
-You can control where hosts in your Shadow experiment get attached to the topology by setting the _iphint_, _citycodehint_, and _countrycodehint_ in your host attribute list in the Shadow config file. Hosts are always attached to the closest match to the best vertex following the hinted restrictions. See [3.1-Shadow-Config.md](3.1-Shadow-Config.md) for more details.
+#### Graph
 
-### Edges
+Shadow represents a network topology over which processes can communicate using a [weighted graph](https://en.wikipedia.org/wiki/Graph_(discrete_mathematics)). The graph contains _vertices_ representing network routers, and _edges_ representing network paths between the routers. Shadow requires that the graph is [connected](https://en.wikipedia.org/wiki/Connectivity_(graph_theory)) such that there exists at least one _path_ (a series of one or more edges) between every pair of vertices.
```

Comment:
One thing that originally confused me about the topology is that the vertices are only sort-of network routers. These vertices are used for determining the path of packets through the network, but attributes you set on these vertices don't act like attributes on a network router. For example if you set a bandwidth for a vertex, it's not actually a bandwidth for a router (since Shadow doesn't simulate routers). It's a default bandwidth for any hosts that are attached to the vertex. Would it be better to refer to them as just "network nodes" or "routing nodes"?